contracts/EventBasedPredictionMarket.sol

```diff
@@ -0,0 +1,333 @@
+        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
+
+        collateralToken.safeApprove(address(optimisticOracle), proposerReward);
+
+        optimisticOracle.requestPrice(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            collateralToken,
+            proposerReward
+        );
+
+        // Set the Optimistic oracle liveness for the price request.
+        optimisticOracle.setCustomLiveness(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleLivenessTime
+        );
+
+        // Set the Optimistic oracle proposer bond for the price request.
+        optimisticOracle.setBond(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData,
+            optimisticOracleProposerBond
+        );
+
+        // Make the request an event-based request.
+        OptimisticOracleInterfaceEventBased(address(optimisticOracle)).setEventBased(
+            priceIdentifier,
+            expirationTimestamp,
+            customAncillaryData
+        );
+
+        priceRequested = true;
+    }
+
+    // Callback function called by the optimistic oracle when a price requested
+    // by this contract is disputed.
+    function priceDisputed(
+        bytes32 identifier,
+        uint256 timestamp,
+        bytes memory ancillaryData,
+        uint256 refund
+    ) external {
+        OptimisticOracleInterface optimisticOracle = _getOptimisticOracle();
+        require(msg.sender == address(optimisticOracle), "not authorized");
+
+        expirationTimestamp = getCurrentTime();
+        require(timestamp <= expirationTimestamp, "dispute and request must have different timestamps");
+        require(identifier == priceIdentifier, "dispute and request must have the same identifier");
+        require(
+            keccak256(ancillaryData) == keccak256(customAncillaryData),
+            "dispute and request must have the same ancillary data"
+        );
+        require(refund == proposerReward, "dispute and request must have the same proposerReward amount");
+
+        _requestOraclePrice();
+    }
+
+    /****************************************
+     *          POSITION FUNCTIONS          *
+     ****************************************/
+
+    /**
+     * @notice Creates a pair of long and short tokens equal in number to tokensToCreate. Pulls the required collateral
+     * amount into this contract, defined by the collateralPerPair value.
+     * @dev The caller must approve this contract to transfer `tokensToCreate * collateralPerPair` amount of collateral.
+     * @param tokensToCreate number of long and short synthetic tokens to create.
+     * @return collateralUsed total collateral used to mint the synthetics.
+     */
+    function create(uint256 tokensToCreate) public requestInitialized nonReentrant returns (uint256 collateralUsed) {
+        require(
```

Comment:
why are you checking this?

Reply:
Oh, I thought I had to prevent people from creating more tokens after the OO price request was fulfilled, but that shouldn't be an issue.